lib/msf/core/post/file.rb

```diff
@@ -354,7 +363,10 @@ def read_file(file_name)
 
     return unless %w[shell powershell].include?(session.type)
 
-    if session.platform == 'windows' || session.platform == 'win'
+    if session.type == 'powershell'
+      return cmd_exec("Get-Content \"#{file_name}\"")
+    end
```

Comment:
I don't think this is going to work for binary files. The one below it for command sessions won't either but the `_read_file_meterpreter` will.

Could you please update this to work with binary files? At the very least that would mean base64 encoding it. If you could also compress it, that would be even better. It would also be worth testing how it works then when the files is quite large like say a gigabyte. Since it'll be more complicated than a single command, you may want to define the logic in a new `_read_file_powershell` method, following the pattern set by the Meterpreter handler.

Reply:
Sure, being able to read binary files would be great but  how about a different PR dedicated for this purpose?